towhee/hub/file_manager.py

```diff
@@ -393,16 +389,19 @@ def get_operator(self, operator: str, branch: str = 'main', redownload: bool = F
 
             if author == 'local':
                 if found_existing is False:
-                    # TODO: filip-halt
-                    # Error logging.
                     engine_log.info('Local file not found, has it been imported?')
-                    file_path = None
+                    raise FileNotFoundError('Local file not found, has it been imported?')
+                return file_path
+
+            if file_path.is_file() is False:
+                download_repo(author, repo, branch, str(file_path.parent), install_reqs=install_reqs)
             else:
-                if redownload:
-                    # TODO: filip-halt
-                    # Error logging.
-                    rmtree(file_path.parent)
+                repo_path = str(path).rsplit('/', 1)[0]
+                repo_name = str(path).rsplit('/', 1)[1].split('.')[0]
+                repo = git.Repo(repo_path)
+                if not 'Your branch is up to date' in repo.git.status():
+                    print(f'Your local operator {repo_name} is not up to date, updating to latest version...')
+                    fetch_head = repo.remote('origin').fetch(branch)
+                    repo.git.reset(fetch_head, '--hard')
```

Comment:
Why use `reset`, if having some changes in local files, they will be lost.

Reply:
fixed